packages/my_at_localization/src/my_at_pose_node.py

```diff
@@ -214,11 +215,12 @@ def detectATPose(self, image_msg):
         detected_tags = self.at_detector.detect(gray_img, estimate_tag_pose=True, camera_params=camera_params, tag_size=0.065)
         detected_tag_ids = list(map(lambda x: fetch_tag_id(x), detected_tags))
         array_for_pub = Int32MultiArray(data=detected_tag_ids)
-        self.tag_pub.publish(array_for_pub)
         for tag_id, tag in zip(detected_tag_ids, detected_tags):
             print('detected {}: ({}, {})'.format(tag_id, image_msg.header.stamp.to_time(), rospy.Time.now().to_time()))
             self._broadcast_detected_tag(image_msg, tag_id, tag)
 
+        self.tag_pub.publish(array_for_pub)
+
```

Comment:
The detected tags should get published after when the TFs are being broadcasted. Otherwise the sensor model would receive the TFs from previous AT Detection.